common/src/main/java/io/pravega/common/util/CompositeByteArraySegment.java

```diff
@@ -183,6 +185,31 @@ public CompositeArrayView slice(int offset, int length) {
         assert length == 0 : "Collection finished but " + length + " bytes remaining";
     }
 
+    @Override
+    public Iterator<ByteBuffer> iterateBuffers() {
+        if (this.length == 0) {
+            return Collections.emptyIterator();
+        }
+
+        AtomicInteger arrayOffset = new AtomicInteger(getArrayOffset(0));
```

Comment:
Why AtomicInteger? What about using some simpler int holder without concurrency support?

Reply:
Because I need to use this in a lambda just below which doesn't like me fiddling around with the variable contents (it needs to be immutable). 

The reason for a lambda is because this method needs to return an iterator, and the simplest way to make one on-the-fly is by using `Arrays.stream()...` and then doing some mapping, hence the need for lambda.